halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/versions/BomVersionCommand.java

```diff
@@ -64,17 +64,25 @@ public String getMainParameter() {
   }
 
   public String getVersion() {
+    String version;
     switch (versions.size()) {
       case 0:
-        return new OperationHandler<String>()
+        version = new OperationHandler<String>()
             .setOperation(Daemon.getVersion(getCurrentDeployment(), false))
             .setFailureMesssage("Failed to get version of Spinnaker configured in your halconfig.")
             .get();
+        break;
       case 1:
-        return versions.get(0);
+        version = versions.get(0);
+        break;
       default:
         throw new IllegalArgumentException("More than one version supplied");
     }
+
+    if (!BomVersionCommand.isProjectVersionFormat(version)) {
```

Comment:
Versions like `master-latest-unvalidated` are perfectly find but would be rejected here. I don't think we want to do the check in this place, but instead when trying to lookup the BOM.

Reply:
This is where the BOM lookup happens: https://github.com/spinnaker/halyard/blob/master/halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/VersionsService.java#L63